### PR TITLE
Structure change

### PR DESCRIPTION
**Pull request by Mihaiii**
Work in progress. Not for merge.

### CODE REVIEW

**Comment by Mihaiii** [comment]
contentstructure

**Comment by Mihaiii** [comment]
After making the previous image, I noticed that "Search Bar page" is redundant. I might include it in Side Panel Page later on.

**Comment by Mihaiii** [comment]
contentstructure

**Comment by Mihaiii** [comment]
To do: remove reference to local Simplified after Simplified's `develop` is merged into `master` and a new RC is made.

**Comment by Mihaiii** [comment]
This is ready for a review.
Checks are failing (at least) because I use a property that's not in Simplified.
The reviewer should either wait until Simplified's `develop` is merged into `master` or reference a local Simplified that uses `develop` branch.

File: src/Content/Api/ContentHandlers.cs [hunk 1]
```diff
@@ -18,34 +19,33 @@ public void Register()
...
-
-                if (item != null)
+                WebContent item = Db.SQL<WebContent>("SELECT i FROM Simplified.Ring6.WebContent i WHERE i.ObjectID = ? and i.Deleted <> ?", query, true).First;
+                if (item != null && item.Url != null)
                 {
-                    return HandleWebContent(item);
-                }
 
-                item = Db.SQL<WebContent>("SELECT i FROM Simplified.Ring6.WebContent i WHERE i.Path = ? and i.Deleted <> ?", url, true).First;
+                    if (IsAllowedToViewContent(item))
```
Comment (habib535) [inline src/Content/Api/ContentHandlers.cs]: Only admin is allowed in this method, generally not everyone is part of the admin group. Is this as-designed?

File: src/Content/Api/CommitHooks.cs [hunk 2]
```diff
@@ -22,29 +23,82 @@ public void Register()
...
+                        {
+                            var wc = Db.SQL<WebContent>("SELECT i FROM Simplified.Ring6.WebContent i WHERE i.NextSiblingKey = ? and i.Deleted <> ?", (a as WebContent).GetObjectID(), true).First;
+                            if (wc != null)
+                            {
+                                wc.NextSiblingKey = (a as WebContent).NextSiblingKey;
+                            }
+                            (a as WebContent).NextSiblingKey = null;
+                        });
+                        editorpage.Transaction.Scope(() => editorpage.CleanCreate());
+                        session.CalculatePatchAndPushOnWebSocket();
```
Comment (habib535) [inline src/Content/Api/CommitHooks.cs]: why do we need to manually call `CalculatePatchAndPushOnWebSocket` on the session? I mean what does this line do?

File: src/Content/Api/CommitHooks.cs [hunk 3]
```diff
@@ -22,29 +23,82 @@ public void Register()
...
-        {
-            MasterPage page = GetMasterPage();
+                var page = session.Store[nameof(MasterPage)] as MasterPage;
 
-            if (page == null)
-            {
-                return;
-            }
-            page.RefreshCurrentPage();
+                page?.Transaction.Scope(() => page.RefreshCurrentPage());
```
Comment (habib535) [inline src/Content/Api/CommitHooks.cs]: Functionality in all the `RefreshCurrentPage` methods is repeating and can be moved to a common utility class

File: src/Content/Api/PartialHandlers.cs [hunk 4]
```diff
@@ -9,119 +9,154 @@ class PartialHandlers : IHandler
...
+                mcontent = new MainContentPage();
+                Session.Current.Store[nameof(MainContentPage)] = mcontent;
             }
-            
-            return treemaster;
+
+            return mcontent;
+        }
+
+        public static SearchResultPage GetSearchResultPageFromSession()
```
Comment (habib535) [inline src/Content/Api/PartialHandlers.cs]: `GetSearchResultPageFromSession` method is not used anywhere, can be removed

File: src/Content/Api/PartialHandlers.cs [hunk 5]
```diff
@@ -9,119 +9,154 @@ class PartialHandlers : IHandler
...
+                {
+                    Session.Current.Store[nameof(EditorPage)] = page;
+                }
+                page.WebContent.Data = DbHelper.FromID(DbHelper.Base64DecodeObjectID(key)) as WebContent;
+
+                var wc = Db.SQL<WebContent>("SELECT i FROM Simplified.Ring6.WebContent i WHERE i.Name IS NOT null and i.Key <> ? and i.Deleted <> ?", key, true);
+
+                foreach (var wc_content in wc)
+                {
+                    if (SkipParent(wc_content.ParentPage, key, wc)) {
```
Comment (habib535) [inline src/Content/Api/PartialHandlers.cs]: `SkipParent` method has a time complexity of 3 nested levels, can it be simplified?

File: src/Content/Content.csproj [hunk 6]
```diff
@@ -75,92 +78,118 @@
     <Compile Include="Helpers\DataHelper.cs" />
```
Comment (habib535) [inline src/Content/Content.csproj]: There is no longer any reference for `DataHelper`, is this for the features you will be adding in the future?

File: src/Content/Api/CommitHooks.cs (hunk 2 above)
Comment (Mihaiii) [reply src/Content/Api/CommitHooks.cs]: In order to send the patch so that the content of the treemenu updates.

**Comment by Mihaiii** [comment]
"Child content move up/down is moving parents, I was expecting it to move within its parent only." Can you make a gif please or give more details? If you try to move a child up/down, it should move within the parent and this is the way I'm aware it currently works.

File: src/Content/Api/ContentHandlers.cs (hunk 1 above)
Comment (Mihaiii) [reply src/Content/Api/ContentHandlers.cs]: If the content item is "Private", then only admins should be able to see it. If it's not private, then everybody can see the content item.

Comment (Mihaiii) [reply src/Content/Api/ContentHandlers.cs]: I didn't got the part about "only admin is allowed". Could you please give more details?

Comment (habib535) [reply src/Content/Api/ContentHandlers.cs]: this is working fine, `content/dynamic` is a view only page and is viewable by any signed-in user while `content/edit` is to make some edits and is only allowed to Admin users

**Comment by Mihaiii** [comment]
> a nice to have thing might be to add Hand cursor on the icons and add some tooltip

I created an issue for this: https://github.com/StarcounterApps/Content/issues/116 . I intend to solve it in another PR.

**Pull request merged**